msprime/demography.py

```diff
@@ -392,6 +392,24 @@ def _check_population_references(self, populations: list[str | int]):
             # the KeyError message
             self[pop_ref]
 
+    def _add_activate_population_event(
+        self, time: float, *, population: str | int
+    ) -> ActivatePopulationEvent:
+        """
+        Activates a population at the specified time. The population is expected to be
+        initially inactive.
+
+        :param float time: The time at which this event occurs in generations.
+        :param str, int population: The population to activate.
+        """
+        self._check_population_references([population])
+        pop = self[population]
+        if pop.initially_active is None:
```

Comment:
It's not clear to me that we want to put in this complexity - we'll only really be using this functionality within Demes models, where we'll be controlling the ``initially_active`` state explicitly anyway, won't we?

Reply:
For now, I imagine it will really only be used in the Demes import interface. But I wasn't sure how much to put into it, in case someone does want to use it outside of Demes in the future.

Reply:
I'd put in the minimum possible semantics for what we need right now - we're not putting it into the public interface, so we don't have to worry about forwards compatibility.